pageserver/src/consumption_metrics/upload.rs

```diff
@@ -268,7 +269,7 @@ impl RawMetricExt for RawMetric {
     }
 }
 
-trait KeyGen<'a>: Copy {
+pub(crate) trait KeyGen<'a> {
     fn generate(&self) -> IdempotencyKey<'a>;
 }
```

Comment:
Is this used from somewhere now?

Reply:
Discussed on Slack, I'll submit a PR tomorrow to remove this trait and just construct the idempotency key directly.